### ISSUE TITLE
Local api update api can't update block with localized

### ISSUE THREAD

**Issue by cmngan** [payloadcms/payload#3189]
### Link to reproduction

https://github.com/cmngan/payload-cms-local-api-update-error

### To Reproduce

1. Use local api, e.g. in a collection, add `endpoints`
<img width="199" alt="image" src="https://github.com/payloadcms/payload/assets/20035088/6cc3f25f-671b-4742-b4c0-59b9cdfd0f93">

2. Inside endpoint, use local api as 
```
handler: async (req, res, next) => {
    const item = {} // `{}` is a placeholder, real data see step 3
    req.payload.update({
        collection: 'some com',
        data: item,
        locale: "all",
        id: item.id,
      })
    )
}
```
3. item is like below(simplified)
```
	{
		"id": "sample",
		"adminTitle": "Example",
		"template_modules_v3": [
			{
				"selectedSites": ["en-kr", "en-gl"],
				"title": {
					"en-gl": "eeeeeeeeeeee"
				},
				"id": "16709298540833613392",
				"blockType": "overview"
			}
		],
		"city": "sydney",
		"author": "638d5ab62d091072b2142a1a",
		"_status": "published"
	}
```
4. Call the endpoint
5. Some data can be updated, e.g. `adminTitle`, `template_modules_v3` -> `selectedSites` 
6. Some data can NOT be updated, e.g. `template_modules_v3` -> `title` -> `en-gl`
7. If block id(16709298540833613392) is changed to something else, title can be updated as well.

### Describe the Bug

See above `To Reproduce`. Seems like it is a bug inside `block` with localized is true. 
Since `selectedSites` is working, I guess localized is related.

### Payload Version

1.13.4

**Comment by AlessioGr** [comment]
Thank you @cmngan !

I've checked it out and, for some reason, I can reproduce that the `localized` field inside the array IS updated, but the top-level `localized` field, on the other hand, is not.

Will do some more digging into this

**Comment by cmngan** [comment]
> Thank you @cmngan !
> 
> I've checked it out and, for some reason, I can reproduce that the `localized` field inside the array IS updated, but the top-level `localized` field, on the other hand, is not.
> 
> So, the exact opposite way.
> 
> Will do some more digging into this

Thanks @AlessioGr 
Interesting finding. 👀

**Comment by JarrodMFlesch** [comment]
@cmngan I believe this has been resolved in 2.0. Could you update and let me know?

**Comment by JarrodMFlesch** [comment]
@cmngan I am going to close this since I was unable to repro on 2.x. Please ping us if it is still a problem for you and we can gather more details!
